### DISCUSSION TITLE
Failed to upgrade legacy queries Datasource hsXuaSe7k was not found

### DISCUSSION THREAD

**Discussion by youssefbenfarhat** [grafana/grafana#53038] [General]
Hello community,

When I export my dashboard and import it again in another server, I get this error. 

I  think the problem is whith the datasource UID. So can I export the dashboard without the UID or is there any another solution to change it when I import it again in another server

**Comment by usmangt** [comment]
Hi @youssefbenfarhat ,

I am closing this issue here on Github and moving it to our Community forums, as it seems more of a general question.

 Please provide some details so that the community members can help you out.

**Comment by Danmdo** [comment]
Hi @youssefbenfarhat , I'm having the same problem has anyone found a solution?

**Comment by logicalsingular** [comment]
@youssefbenfarhat, @Danmdo, just edit json file. Replace old UID to new datasource UID. That solves your issue.

**Comment by miguelHM63** [comment]
I had the same problem, I didn't know where to get the uid of my data source, so I had to review the request that grafana made to see what information it brought when listing the data sources and I found the valuable UID
![image](https://user-images.githubusercontent.com/30902350/194108145-ae903725-4f61-4841-932e-656572256e27.png)

**Reply by anderfrank** [reply]
So glad I found this, had the same problem when importing a backup of one of my dashboards. THANK YOU!

**Reply by jcquiles** [reply]
@MiguelHuamanMedina what is the process involved in locating that the UID of the datasource? Looks like developer tools in the browser.

**Reply by wajahatlateef** [reply]
@MiguelHuamanMedina will you please explain steps to get uid and where to replace it?

**Reply by marko-hologram** [reply]
@jcquiles @wajahatlateef open `/datasources` in your Grafana dashboard. You can do that by clicking Settings -> Data sources or just typing that URL in your browser. Open your browser dev tools (F12) and go to Network tab. Filter XHR (AJAX) requests and find the "datasources" request. Now open response tab for that request and find your datasource's uid. If you have multiple datasources, just find the appropriate one in that data.

![image](https://user-images.githubusercontent.com/22904204/212773522-ede6c4b3-644d-4079-89a5-35d29efee8dc.png)

> I erased my uid here, but you get the point.

Then open your problematic dashboard and click "Dashboard Settings" on top (small icon). Then click "JSON Model" on the left and search (Ctrl + F) for "uid" property that matches the uid reported by the error message you get in Grafana. Replace that uid value with the one you got from dev tools previously and that's it. This is what worked for me just few minutes ago.

Hopefully this helps!

**Reply by miguelHM63** [reply]
STEP 1
![image](https://user-images.githubusercontent.com/30902350/212800832-b95264ee-c0bb-4e40-b889-d12e94a29b6d.png)
*****

STEP 2 - with developer tools open (F12) click on the request you made to bring up the data sources
![image](https://user-images.githubusercontent.com/30902350/212801193-9583817b-01bd-466c-b782-e8b5bc384ce7.png)
*****

STEP3 - click on the response tab and you will see all your datasources with their respective uid
![image](https://user-images.githubusercontent.com/30902350/212804200-c79d91ec-b38f-479b-882d-0df04bbb914c.png)

**Comment by mm-52tFd** [comment]
I found another solution. In my case the Postgres data source was named incorrectly. In Grafana, go to settings and make sure that the teslamate database has the right name. In the imported json dashboard definitions, I noticed that the datasource was named `TeslaMate`. I had named the postgres data source `Teslamate`. When I updated the name to have proper capitalization to match the dashboard json, it connected immediately and works fine now :)

If the datasource has to be named something different, then if you update the datasource name in the json for each dashboard, they should then work.

**Reply by jacxon4** [reply]
I had the same stupid issue for 2 days... Came here from the hassio Teslamate integration and you solve it. Thanks!

**Comment by natellium** [comment]
Hey there, I believe you're running into the same issue as [this forum post](https://community.grafana.com/t/failed-to-upgrade-legacy-queries-datasource/106291). If you want to use the dashboard in a different instance, you should select `Export for sharing externally`, and you'll be able to select the DS you want to use when importing the dashboard. If the issue persists, please open a bug [here](https://github.com/grafana/grafana/issues/new/choose) :)
